### ISSUE TITLE
Subtitle custom format support

### ISSUE THREAD

**Issue by Maxmystere** [Radarr/Radarr#9707]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

I would like to be able to do the same as the current LanguageSpecification but for subtitles

### Describe the solution you'd like

Adding a new `CustomFormatSpecificationBase` that would read the `input.Movie.MovieFile.MediaInfo.Subtitles`

### Describe alternatives you've considered

Using the Release Title condition but it isn't really practical

### Anything else?

.

**Comment by Maxmystere** [comment]
A practical usage would be to ensure the file has the subtitles embedded (for example mkv file)

**Comment by Qstick** [comment]
The problem is, we won't know what it is 50% of the time prior to downloading. So upgrades will never get grabbed reliably.

**Comment by Maxmystere** [comment]
From what I saw this is what the Language Custom Format already does no ?

**Comment by Qstick** [comment]
Language is in the release title much more often than subtitle language is

**Comment by Maxmystere** [comment]
You can tell me if I'm wrong with anything
After checking a bit more the LanguageSpecification class https://github.com/Radarr/Radarr/blob/4a9c0b2240ebe53f759db4409c39dfa59efbaf70/src/NzbDrone.Core/CustomFormats/Specifications/LanguageSpecification.cs#L23
It's detecting the language before downloading then also after with the resulting downloaded file 
Before:
https://github.com/Radarr/Radarr/blob/4a9c0b2240ebe53f759db4409c39dfa59efbaf70/src/NzbDrone.Core/CustomFormats/CustomFormatCalculationService.cs#L44

After:
https://github.com/Radarr/Radarr/blob/4a9c0b2240ebe53f759db4409c39dfa59efbaf70/src/NzbDrone.Core/CustomFormats/CustomFormatCalculationService.cs#L207

So if I understood well the issue with my proposal is that there is no "before" checkup ?

My goal was to improve the already existing basic subtitle detection possible with [trash-guide](https://trash-guides.info/Radarr/Radarr-collection-of-custom-formats/#vostfr) and make it possible to invalidate a downloaded files to try to find a better match if no subtitles were found

**Comment by Qstick** [comment]
Correct, very rarely do we know the subtitle language BEFORE download. This results in an issue that the file on disk will always be scored higher than any release on the indexer (simply due to us not knowing the sub languages). It's hard to implement a custom format specification on a property where the BEFORE and AFTER are not each reliable.

**Comment by Maxmystere** [comment]
My idea for using this Custom format was to give a negative score if no good subtitles were found
For example it would give a score of 0 before downloading then after downloading would either give a +1 or -1 to the resulting file
Is this a relevant use case for the Custom Formats ?
